src/main/java/com/synopsys/integration/jenkins/scan/service/BridgeDownloadParametersService.java

```diff
@@ -63,23 +103,60 @@ public String getBridgeDownloadUrlFromGlobalConfig() {
-        } else {
-            String bridgeDownloadUrlFromGlobalConfig = getBridgeDownloadUrlFromGlobalConfig();
-            if (bridgeDownloadUrlFromGlobalConfig != null) {
-                bridgeDownloadParameters.setBridgeDownloadUrl(bridgeDownloadUrlFromGlobalConfig);
-            }
+            bridgeDownloadParameters.setBridgeDownloadUrl(
+                    scanParameters.get(ApplicationConstants.BRIDGE_DOWNLOAD_URL).toString().trim());
         }
+        else if (getBridgeDownloadUrlFromGlobalConfig() != null) {
+            bridgeDownloadParameters.setBridgeDownloadUrl(getBridgeDownloadUrlFromGlobalConfig());
+        }
+        else if (scanParameters.containsKey(ApplicationConstants.BRIDGE_DOWNLOAD_VERSION)) {
+            String desiredVersion = scanParameters.get(ApplicationConstants.BRIDGE_DOWNLOAD_VERSION).toString().trim();
+            String bridgeDownloadUrl = String.join("/", ApplicationConstants.BRIDGE_ARTIFACTORY_URL,
+                    desiredVersion, getSynopsysBridgeZipFileName(desiredVersion));
 
-        if (scanParameters.containsKey(ApplicationConstants.BRIDGE_DOWNLOAD_VERSION)) {
-            bridgeDownloadParameters.setBridgeDownloadVersion(scanParameters.get(ApplicationConstants.BRIDGE_DOWNLOAD_VERSION).toString());
+            bridgeDownloadParameters.setBridgeDownloadUrl(bridgeDownloadUrl);
+            bridgeDownloadParameters.setBridgeDownloadVersion(desiredVersion);
+        }
+        else {
+            String bridgeDownloadUrl = String.join("/", ApplicationConstants.BRIDGE_ARTIFACTORY_URL,
+                    ApplicationConstants.SYNOPSYS_BRIDGE_LATEST_VERSION, getSynopsysBridgeZipFileName());
+            bridgeDownloadParameters.setBridgeDownloadUrl(bridgeDownloadUrl);
         }
+        return bridgeDownloadParameters;
+    }
 
-        if (scanParameters.containsKey(ApplicationConstants.BRIDGE_INSTALLATION_PATH)) {
-            bridgeDownloadParameters.setBridgeInstallationPath(scanParameters.get(ApplicationConstants.BRIDGE_INSTALLATION_PATH).toString());
+    public String getPlatform() {
+        String os = null;
+
+        if (workspace.isRemote()) {
+            try {
+                os = workspace.act(new OsNameTask());
+            } catch (IOException | InterruptedException e) {
+                listener.getLogger().println("An exception occurred while fetching the OS information for the agent node: " + e.getMessage());
+            }
+        } else {
+            os = System.getProperty("os.name").toLowerCase();
         }
 
-        return bridgeDownloadParameters;
+        if (os.contains("win")) {
+            return ApplicationConstants.PLATFORM_WINDOWS;
+        } else if (os.contains("mac")) {
+            return ApplicationConstants.PLATFORM_MAC;
+        } else {
+            return ApplicationConstants.PLATFORM_LINUX;
+        }
+    }
+
+    public String getSynopsysBridgeZipFileName() {
+        return "synopsys-bridge-".concat(getPlatform()).concat(".zip");
```

Comment:
synopsys-bridge name looks to be used at multiple places. Shall we make it as constant.

Reply:
Updated - https://github.com/synopsys-sig/synopsys-security-scan-plugin/blob/dev/src/main/java/com/synopsys/integration/jenkins/scan/service/BridgeDownloadParametersService.java#L157